utils/utils.hpp

```diff
@@ -329,5 +329,28 @@ class Utils
       mlpack::Log::Warn << "The " << path << " doesn't exist." << std::endl;
     }
   }
+
+  /**
+   * Loads a vocabulary file and stores the content into a vector.
+   *
+   * @param vocabPath The path to the vocabulary file.
+   * @param vocabulary Stores the vocabulary content along with indices.
+   */
+  void LoadVocabulary(const std::string vocabPath,
+                      std::map<std::string, size_t>& vocabulary)
```

Comment:
I think `std::unordered_map` could be faster.
```suggestion
                      std::unordered_map<std::string, size_t>& vocabulary)
```